operator/storage/config_lock.go

```diff
@@ -22,5 +23,13 @@ func (stored *ConfigLock) ValidateCompatibility(current *ConfigLock) error {
 		return fmt.Errorf("enabling local events is not allowed. The database must be removed or reinitialized")
 	}
 
+	if stored.UsingSSVSigner && !current.UsingSSVSigner {
+		return fmt.Errorf("disabling ssv-signer is not allowed. The database must be removed or reinitialized")
+	}
+
+	if !stored.UsingSSVSigner && current.UsingSSVSigner {
+		return fmt.Errorf("enabling ssv-signer is not allowed. The database must be removed or reinitialized")
+	}
```

Comment:
can we maybe do something simpler? wdyt? @moshe-blox @nkryuchkov 

```
if stored.UsingSSVSigner != current.UsingSSVSigner {
    return fmt.Errorf("changing ssv-signer isn't allowed. stored: %v, current: %v, database must be reinitailzed", stored.UsingSSVSigner, current.UsingSSVSigner)
}
```

Reply:
Maybe but I did it similarly to what we do with `UsingLocalEvents`. Having both checks allows to test both cases, so IMO it's ok